src/Instruments.jl

```diff
@@ -28,59 +28,65 @@ currency(::Instrument{S,Currency{CS}}) where {S,CS} = currency(CS)
 `Position` represents ownership of a certain quantity of a particular financial instrument.
 """
 struct Position{I<:Instrument,A}
-    instrument::I
     amount::A
 end
-# Position(inst::I,a::A) where {I<:Instrument,A<:Real} = Position{I,A}(inst,a)
+Position{I}(a) where {I<:Instrument} = Position{I,typeof(a)}(a)
 
 if VERSION >= v"1.3"
-    (instrument::Instrument)(amount) = Position(instrument,amount)
+    (instrument::Instrument)(amount) = Position{typeof(instrument)}(amount)
 end
 
 """
-Returns the financial instrument (as an instance) for a position.
+Returns the financial instrument type of a position.
 """
-instrument(p::Position) = p.instrument
+instrument(::Position{I}) where {I} = I()
 
 """
 Returns the amount of the instrument in the `Position` owned.
 """
 amount(p::Position) = p.amount
 
+"""
+Returns the symbol of the instrument in the `Position`.
+"""
+symbol(::Position{I,A}) where {I,A} = symbol(I())
```

Comment:
You should be able to do this directly on the type, without having to create an instance, and without calling `symbol` either.
Has symbol been imported from Base?  This really should be `Base.symbol`